x/protorev/keeper/epoch_hook.go

```diff
@@ -0,0 +1,117 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	epochstypes "github.com/osmosis-labs/osmosis/v13/x/epochs/types"
+	"github.com/osmosis-labs/osmosis/v13/x/protorev/types"
+)
+
+type EpochHooks struct {
+	k Keeper
+}
+
+// Struct used to track the pool with the highest liquidity
+type LiquidityPoolStruct struct {
+	Liquidity sdk.Int
+	PoolId    uint64
+}
+
+var (
+	_ epochstypes.EpochHooks = EpochHooks{}
+)
+
+func (k Keeper) EpochHooks() epochstypes.EpochHooks {
+	return EpochHooks{k}
+}
+
+// BeforeEpochStart is the epoch start hook.
+func (h EpochHooks) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	return nil
+}
+
+// AfterEpochEnd is the epoch end hook. The module will update all of the pools in the store that are
+// used for trading.
+func (h EpochHooks) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) error {
+	switch epochIdentifier {
+	case "week":
+		// Reset the pools in the store
+		h.k.DeleteAllAtomPools(ctx)
+		h.k.DeleteAllOsmoPools(ctx)
+
+		// Update the pools in the store
+		return h.k.UpdatePools(ctx)
+	}
+	return nil
+}
+
+func (k Keeper) UpdatePools(ctx sdk.Context) error {
```

Comment:
Asking to learn: When we fix arb and perform swaps who'll be paying for swap fees? Would it be beneficial to route swap between high fee vs low fee pools?

Reply:
The module account will be minting the coins and paying for the swap fees. If swap fees end up consuming the entire arb, the module won't execute that trade (we simulate the trade before executing it). This will be more hashed out in PR #3 and #4 (https://github.com/osmosis-labs/osmosis/issues/3594). 

Building routes with low fee vs high fee pools is an interesting point. I think @NotJeremyLiu might have some more insight into that.

Reply:
Module account pays for the swap (module account just calls MultiHopSwapExactAmountIn like a normal account would).

Determining the optimal input amount accounts for swap fees.

If liquidity, weight, and assets were equal on 2 pools, routing through lower fee pools would be beneficial.